app/Http/Controllers/BiosController.php

```diff
@@ -0,0 +1,144 @@
+    }
+
+    public function show($id)
+    {
+        try {
+            $bio = Bio::where('id', $id)->firstOrFail();
+        } catch (Illuminate\Database\Eloquent\ModelNotFoundException $e) {
+            Session::flash('error-message', 'Sorry, but that isn\'t a valid URL.');
+            Log::error($e);
+            return Redirect::to('/');
+        }
+
+        if ($bio->user_id != Auth::user()->id) {
+            Session::flash('error-message', 'Sorry, but you don\'t own that bio.');
+            Log::error('User ' . Auth::user()->id . ' tried to view a bio they don\'t own.');
+            return Redirect::to('/');
+        }
+
+        return View::make('bios.show')
+            ->with('bio', $bio);
+    }
+
+    public function edit($id)
+    {
+        try {
+            $bio = Bio::where('id', $id)->firstOrFail();
+        } catch (Illuminate\Database\Eloquent\ModelNotFoundException $e) {
+            Session::flash('error-message', 'Sorry, but that isn\'t a valid URL.');
+            Log::error($e);
+            return Redirect::to('/');
+        }
+
+        if ($bio->user_id != Auth::user()->id) {
+            Session::flash('error-message', 'Sorry, but you don\'t own that bio.');
+            Log::error('User ' . Auth::user()->id . ' tried to edit a bio they don\'t own.');
+            return Redirect::to('/');
+        }
+
+        return View::make('bios.edit')
+            ->with('bio', $bio);
+    }
+
+    public function update($id)
+    {
+        $data = Input::all();
+
+        // Make validator
+        $validator = Validator::make($data, $this->validation);
+
+        if ($validator->passes()) {
+            // Pull
+            $bio = Bio::where('id', $id)->firstOrFail();
+
+            // Validate ownership
+            if ($bio->user_id != Auth::user()->id) {
+                Session::flash('error-message', 'Sorry, but you don\'t own that bio.');
+                Log::error('User ' . Auth::user()->id . ' tried to edit a bio they don\'t own.');
+                return Redirect::to('/');
+            }
+
+            // Save
+            $bio->body = Input::get('body');
+            $bio->user_id = Auth::user()->id;
+            $bio->save();
+
+            Session::flash('message', 'Successfully edited bio.');
+
+            return Redirect::to('bios/' . $bio->id);
+        }
+
+        return Redirect::to('bios/' . $id . '/edit')
+            ->withErrors($validator)
+            ->withInput();
+    }
+
+    public function destroy($id)
+    {
+        $bio = Bio::where('id', $id)->firstOrFail();
+
+        // Validate ownership
+        if ($bio->user_id != Auth::user()->id) {
+            Session::flash('error-message', 'Sorry, but you don\'t own that bio.');
+            Log::error('User ' . Auth::user()->id . ' tried to delete a bio they don\'t own.');
+            return Redirect::to('/');
+        }
+
+        Session::flash('success-message', 'Bio successfully deleted.');
+        $bio->delete();
```

Comment:
Maybe flash after instead of before? No _great_ reason, could argue you would want to check the return of `delete` to make sure it deleted, or avoid flashing the message if there was some bizarre database failure that threw an exception, but really it just bothers me in an OCD sort of way to say that it was deleted before you delete it, haha...